src/urlshortner/url_shortner_combined.py

```diff
@@ -0,0 +1,186 @@
+import json
+import hashlib
+import re
+import os
+import concurrent.futures
+from urllib.parse import urlparse
+import requests
+import boto3
+
+
+# Keep tab of global blacklists
+GLOBAL_BLACKLIST = set()
+
+# Override AWS Region below, otherwise will use Lambda function's region
+try:
+    REGION_NAME = os.environ["AWS_REGION"]
+except (NameError, KeyError):
+    REGION_NAME = "us-west-2"
+
+# Set DynamoDB client and table name
+DDB = boto3.client("dynamodb", region_name=REGION_NAME)
+TABLENAME = "url-shortner-new"
+
+# Iterate through the list to locate blacklisted domains
+URLS = ["https://raw.githubusercontent.com/hectorm/hmirror/master/data/spam404.com/list.txt",
+        "https://raw.githubusercontent.com/chadmayfield/pihole-blocklists/master/lists/pi_blocklist_porn_top1m.list",
+        "https://www.stopforumspam.com/downloads/toxic_domains_whole.txt"]
+
+
+def update_blacklist(url):
+    """Update global blacklist from the blacklisted domains."""
+    try:
+        str_lists = requests.get(url).text
+        lists = re.split(r"\n", str_lists)
+        GLOBAL_BLACKLIST.update(set(lists))
+    except requests.exceptions.MissingSchema:
+        pass  # skip invalid urls
+
+
+# Start multithreading on individual urls
+with concurrent.futures.ThreadPoolExecutor() as executor:
+    _ = [executor.submit(update_blacklist, url) for url in URLS]
+
+
+def lambda_handler(event, context):
+    """Call Lambda's function handler."""
+    # /lambda endpoint on API Gateway will keep the function warm
+    if event["path"] == "/lambda":
+        return {
+            "statusCode": 200,
+            "headers": {
+                "Access-Control-Allow-Origin": "*"
+            },
+            "body": json.dumps("Container Warm..")
+        }
+    # "/" resource on API Gateway is the url shortner domain
+    if event["resource"] == "/":
+        response = create_url(event)
+    else:
+        response = get_url(event)
+
+    return response
+
+
+def create_url(event):
+    """Create shortened url from the event."""
+    long_url = event["headers"]["url"]
+    url_components = urlparse(long_url)
+    protocol = url_components.scheme
+
+    # Clean string to remove any whitespace and protocols
+    new_str = long_url.strip()
+    new_str = new_str.replace(" ", "%20")
+    new_str = new_str.replace("https://", "")
+    new_str = new_str.replace("http://", "")
+
+    if protocol:
+        long_url = f"{protocol}://{new_str}"
+    else:
+        long_url = f"http://{new_str}"
+
+    # Check if webpage is legit
+    url_validity = check_valid_url(long_url)
+    print(f"url_validity is {url_validity}")
+
+    if url_validity is not True:
+        return {
+            "statusCode": 403,
+            "headers": {
+                "Access-Control-Allow-Origin": "*"
+            },
+            "body": json.dumps(str(url_validity))
+        }
+
+    short_url = hashlib.md5(long_url.encode("utf-8")).hexdigest()[:6]
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The hashing algorithm you are using is unsecure and might lead to cryptographic vulnerabilities. [Learn more](https://en.wikipedia.org/wiki/Hash_function_security_summary). Use a more secure hashing algorithm when creating message digests using hashlib. For example, use SHA224, SHA256, SHA384, or SHA512. 

[Learn more](https://docs.python.org/3/library/hashlib.html#hash-algorithms)